samples/cpp/tickmeter.cpp

```diff
@@ -0,0 +1,31 @@
+//TickMeter sample program
+#include "opencv2/opencv.hpp"
+#include <iostream>
+
+using namespace cv;
+using namespace std;
+int main(int argc, char** argv)
+{
+    VideoCapture cap;
+    if(argc>1)
+    {
+        cap.open(argv[1]);
+    }
+    else
+    {
+        cap.open(0);
+    }
+    Mat frame;
+    TickMeter t;
+    namedWindow("TickMeter",1);
+    while(true)
+    {
+        t.start();
+        cap>>frame;
+        imshow("TickMeter",frame);
+        waitKey(1);
+        t.stop();
+        cout<<"Current FPS = "<<t.getFps()<<endl;
```

Comment:
> Current FPS

This is not true.

You calculate here an average FPS for all processed frames.

BTW, This metric is not very useful:
- long initialization of the first frame (buffers allocation/initialization) affects metric for the large number of the next frames
- in classical object detection apps processing time depends on number of objects on the input frame

Usually FPS is counted as an average for the last N frames:
- N=1: "true" value of the last processed frame (this is "current FPS"). Cons of this approach is very non-stable value (it changes very quickly, so it is not usable with UI).
- N=10: more stable result; corresponding to the contents of current frames instead of "global" average (for real-time processing app (like 20 FPS), lag is about 0.5 sec).

Reply:
Thanks for your feeback , but if do it other way , we can calculate total frames in a second and print that each second
The only issue is that than fps will be provided at an interval of 1 second so if some process completes in time less than a second it will don't give us any result so I used this method , but I think you are right , the metric is not so useful , so I should think more for some better solution

Reply:
Sometimes algorithms can process several frames at the same time. I mean between `tm.start()` and `tm.stop()` we can process more than one frame. In example, deep learning networks can work with a batch of frames (a time series or just to improve a whole efficiency). However `counter` is incremented just at `tm.stop()`.

Reply:
I think 2 solutions are possible : 
1. increment counter each time frame is read from VideoCapture , and not at the time of calling t.stop() but the issue will be then that if frames are read from any other source for example an array of Mat then it will become senseless.
2.add a public function called increaseCounter() in TickMeter class and then user can independently call it . 

I think solution 2 might work also I will change the algorithm to take average over 10 frames as told by @alalek